### PR TITLE
Simplify `Regex` type and rename it to `PreventSyncPattern`

### PR DESCRIPTION
**Pull request by touilleMan**

### CODE REVIEW

File: libparsec/crates/types/src/prevent_sync_pattern.rs [hunk 1]
```diff
@@ -0,0 +1,125 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use std::{collections::HashSet, fmt::Display};
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum PreventSyncPatternError {
+    #[error("Regex parsing error: {err}")]
+    BadRegex { err: regex::Error },
```
[the remark this answers is not in the document]
Comment (touilleMan) [reply libparsec/crates/types/src/prevent_sync_pattern.rs]: That's a very good point ! thx

File: libparsec/crates/client/src/workspace/store/prevent_sync_pattern.rs [hunk 2]
```diff
@@ -13,7 +13,7 @@ pub enum ApplyPreventSyncPatternError {
 pub(super) async fn ensure_prevent_sync_pattern_applied_to_wksp(
     storage: &mut WorkspaceStorage,
     device: Arc<LocalDevice>,
-    pattern: &Regex,
+    pattern: &PreventSyncPattern,
```
Comment (AureliaDolo) [inline libparsec/crates/client/src/workspace/store/prevent_sync_pattern.rs:16]: nit: rename to prevent_sync_pattern for consistency

File: libparsec/crates/types/src/prevent_sync_pattern.rs [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+            .map(regex_from_glob_pattern)
+            .collect::<Result<_, _>>()?;
+        Ok(Self(regexes))
+    }
+
+    pub fn is_match(&self, string: &str) -> bool {
+        self.0.iter().any(|r| r.is_match(string))
+    }
+
+    /// Create an empty regex, that regex will never match anything
```
Comment (AureliaDolo) [inline libparsec/crates/types/src/prevent_sync_pattern.rs:83]:
```suggestion
    /// Create an empty PreventSyncPattern, that PreventSyncPattern will never match anything
```

**Review by vxgmichel — APPROVED** [review]
LGTM :+1:

**Review by AureliaDolo — APPROVED** [review]

File: libparsec/crates/types/src/prevent_sync_pattern.rs (hunk 3 above)
Comment (touilleMan) [reply libparsec/crates/types/src/prevent_sync_pattern.rs]: :+1:  done.

**Pull request merged**